uniffi_core/src/ffi/foreignexecutor.rs

```diff
@@ -39,19 +39,40 @@ unsafe impl Sync for ForeignExecutorHandle {}
 /// bindings should release the reference to the executor that was reserved for Rust.
 ///
 /// This callback can be invoked from any thread, including threads created by Rust.
+///
+/// The callback should return one of the `FOREIGN_EXECUTOR_CALLBACK_*` codes;
 pub type ForeignExecutorCallback = extern "C" fn(
     executor: ForeignExecutorHandle,
     delay: u32,
     task: Option<RustTaskCallback>,
     task_data: *const (),
-);
+) -> i8;
+
+/// Returned by `ForeignExecutorCallback` if the callback was scheduled successfully
+pub const FOREIGN_EXECUTOR_CALLBACK_SUCCESS: i8 = 0;
+/// Returned by `ForeignExecutorCallback` if the callback couldn't be scheduled because the
+/// foreign executor is canceled/closed.
+pub const FOREIGN_EXECUTOR_CALLBACK_CANCELED: i8 = 1;
+/// Returned by `ForeignExecutorCallback` for all other errors
+pub const FOREIGN_EXECUTOR_CALLBACK_ERROR: i8 = 2;
 
 // Option<RustTaskCallback> should use the null pointer optimization and be represented in C as a
 // regular pointer.  Let's check that.
 static_assertions::assert_eq_size!(usize, Option<RustTaskCallback>);
 
 /// Callback for a Rust task, this is what the foreign executor invokes
-pub type RustTaskCallback = extern "C" fn(*const ());
+///
+/// The task will be passed the `task_data` passed to `ForeignExecutorCallback` in addition to one
+/// of the `RUST_TASK_CALLBACK_*` codes.
+pub type RustTaskCallback = extern "C" fn(*const (), i8);
+
+/// Passed to a RustTaskCallback function when the executor successfully invokes them
+pub const RUST_TASK_CALLBACK_SUCCESS: i8 = 0;
```

Comment:
I'm wondering if that would not be better to have a:

```rust
#[repr(u8)]
pub enum RustTaskCallbackResult {
    Success = 0,
    Canceled = 1,
}
```

Thoughts?

Reply:
Seems good to me.  I used constants mostly just to match the other FFI code, but I don't think it's important to do that.